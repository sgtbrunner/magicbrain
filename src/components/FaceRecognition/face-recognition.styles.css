.bounding-box {
  position: absolute;
  box-shadow: inset 0 0 0 3px #149df2;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -ms-flex-pack: center;
  justify-content: center;
}

.frame {
  /* border: 3px solid rgb(150, 90, 250); */
  margin-bottom: 30px;
}
