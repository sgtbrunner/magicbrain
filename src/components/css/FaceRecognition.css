.bounding-box {
    position: absolute;
    box-shadow: inset 0 0 0 3px #149df2;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -ms-flex-pack: center;
    justify-content: center;
    cursor: pointer;	
}

.frame {
	/*border: 2vh solid white;*/
	margin-bottom:  30px;
}