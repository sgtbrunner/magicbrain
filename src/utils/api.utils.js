const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
