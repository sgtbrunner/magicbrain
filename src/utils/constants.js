export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
