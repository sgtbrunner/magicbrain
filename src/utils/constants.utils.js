const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
