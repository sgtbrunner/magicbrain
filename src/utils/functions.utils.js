export const getUserFromLocalStorage = () => localStorage.getItem('user');
