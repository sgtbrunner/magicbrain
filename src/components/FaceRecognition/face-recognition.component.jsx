import React from 'react';
import PropTypes from 'prop-types';

import ErrorDisplay from '../ErrorDisplay';

import './face-recognition.styles.css';

const FaceRecognition = ({ error, box, imageUrl }) => (
  <div className="centered ma">
    <div className="absolute mt2">
      {error ? (
        <ErrorDisplay error={error} />
      ) : (
        <img id="inputimage" alt="" src={imageUrl} width="500px" height="auto" className="frame" />
      )}
      <div
        className="bounding-box"
        style={{ top: box.topRow, right: box.rightCol, bottom: box.bottomRow, left: box.leftCol }}
      />
    </div>
  </div>
);

FaceRecognition.propTypes = {
  box: PropTypes.shape({
    topRow: PropTypes.number,
    bottomRow: PropTypes.number,
    rightCol: PropTypes.number,
    leftCol: PropTypes.number,
  }).isRequired,
  error: PropTypes.string,
  imageUrl: PropTypes.string.isRequired,
};

FaceRecognition.defaultProps = {
  error: null,
};

export default FaceRecognition;
