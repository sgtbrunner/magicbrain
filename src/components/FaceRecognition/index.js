import FaceRecognition from './face-recognition.component';

export default FaceRecognition;
