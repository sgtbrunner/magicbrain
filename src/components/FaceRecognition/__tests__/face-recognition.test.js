import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import FaceRecognition from '..';

const BOX = { bottomRow: 50, leftCol: 50, rightCol: 50, topRow: 50 };
const ERROR = 'Oooops, there is an error';
const IMG = 'img';

const mountFaceRecognitionComponent = (newProps) =>
  mountComponent({ component: FaceRecognition, props: { imageUrl: '', ...newProps } });

describe('FaceRecognition component', () => {
  it('should match snapshot whilst face location is loading', () => {
    const { container } = mountFaceRecognitionComponent({ loading: true });

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot once face location is loaded', () => {
    const { container } = mountFaceRecognitionComponent({
      loading: false,
      box: BOX,
    });

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot if there is an error', () => {
    const { container } = mountFaceRecognitionComponent({
      loading: false,
      error: ERROR,
    });

    expect(container).toMatchSnapshot();
  });

  it('should render image and spinner whilst loading', () => {
    mountFaceRecognitionComponent({ loading: true });

    const image = screen.getByRole(IMG);
    const spinner = screen.getByTestId('overlay');

    expect(image).toBeInTheDocument();
    expect(spinner).toBeInTheDocument();
  });

  it('should render image and box location once loaded', () => {
    mountFaceRecognitionComponent({
      loading: false,
      box: BOX,
    });

    const image = screen.getByRole(IMG);
    const box = screen.getByTestId('face-location');

    expect(image).toBeInTheDocument();
    expect(box).toBeInTheDocument();
  });

  it('should NOT render image but render ErrorDisplay component if there is an error', () => {
    mountFaceRecognitionComponent({
      loading: false,
      error: ERROR,
    });

    const image = screen.queryByRole(IMG);
    const error = screen.getByRole('alert');

    expect(image).not.toBeInTheDocument();
    expect(error).toBeInTheDocument();
  });
});
