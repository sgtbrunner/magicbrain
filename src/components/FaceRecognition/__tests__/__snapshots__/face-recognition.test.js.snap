// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceRecognition component should match snapshot if there is an error 1`] = `
<div>
  <div
    class="centered"
  >
    <div
      class="absolute mt2"
    >
      <div
        class="mb2"
      >
        <div
          class="_loading_overlay_wrapper css-79elbk"
          data-testid="wrapper"
        >
          <div
            class="bounding-box"
            data-testid="face-location"
          />
        </div>
      </div>
      <p
        class="red f6 mv3 pa3 right-0 left-0 b bg-light-pink ba b--red br2"
        role="alert"
      >
        Oooops, there is an error
      </p>
    </div>
  </div>
</div>
`;

exports[`FaceRecognition component should match snapshot once face location is loaded 1`] = `
<div>
  <div
    class="centered"
  >
    <div
      class="absolute mt2"
    >
      <div
        class="mb2"
      >
        <div
          class="_loading_overlay_wrapper css-79elbk"
          data-testid="wrapper"
        >
          <img
            alt=""
            id="input-image"
            src=""
            width="460"
          />
          <div
            class="bounding-box"
            data-testid="face-location"
            style="top: 50px; right: 50px; bottom: 50px; left: 50px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FaceRecognition component should match snapshot whilst face location is loading 1`] = `
<div>
  <div
    class="centered"
  >
    <div
      class="absolute mt2"
    >
      <div
        class="mb2"
      >
        <div
          class="_loading_overlay_wrapper _loading_overlay_wrapper--active css-ft4slc"
          data-testid="wrapper"
        >
          <div
            class="_loading_overlay_overlay css-df17o1"
            data-testid="overlay"
          >
            <div
              class="_loading_overlay_content css-42igfv"
            >
              <div
                class="_loading_overlay_spinner css-50etie"
              >
                <svg
                  viewBox="25 25 50 50"
                >
                  <circle
                    cx="50"
                    cy="50"
                    fill="none"
                    r="20"
                    stroke-miterlimit="10"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
          <img
            alt=""
            id="input-image"
            src=""
            width="460"
          />
          <div
            class="bounding-box"
            data-testid="face-location"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
